src/shape.rs

```diff
@@ -1080,13 +1080,19 @@ impl ShapeLine {
                             word_range_width += word_width;
                             continue;
                         } else if wrap == Wrap::Glyph
-                            // Make sure that the word is able to fit on it's own line, if not, fall back to Glyph wrapping.
+                            // Make sure that the word is able to fit on its own line, if not, fall back to Glyph wrapping.
                             || (wrap == Wrap::WordOrGlyph && word_width > line_width)
+                            // If we're on the first line
+                            || (wrap == Wrap::WordOrGlyph && visual_lines.len() == 0 &&
+                            // and we can't fit this word on the first line on its own
+                            current_visual_line.w + word_width > line_width)
                         {
                             // Commit the current line so that the word starts on the next line.
                             if word_range_width > 0.
-                                && wrap == Wrap::WordOrGlyph
-                                && word_width > line_width
+                                && ((wrap == Wrap::WordOrGlyph && word_width > line_width)
+                                    || (wrap == Wrap::WordOrGlyph
+                                        && visual_lines.len() == 0
+                                        && current_visual_line.w + word_width > line_width))
```

Comment:
Wondering if storing this in a variable and using it across the 4 instances would be helpful 🤔 . Though, downside is that we'd compute it unnecessarily in the case of going into the first if-block maybe.

i.e.
```
word_exceeds_first_line_available_space = (wrap == Wrap::WordOrGlyph
                                        && visual_lines.len() == 0
                                        && current_visual_line.w + word_width > line_width)
```

Might just be simpler to keep it in-line though - I think the comments are very helpful!

Reply:
(not the best name perhaps above 😅 )

Reply:
Done, thank you!